### PR TITLE
[PR] Remove undefined edge label placement

### LINKED ISSUE #666: Remove undefined edge label placement
We currently default to `EdgeLabelPlacement.UNDEFINED`, which is pretty certainly not what people want. While it does exclude a label from layout, so should the _No Layout_ option, so that's no reason to keep that default. Thus:

* Remove `EdgeLabelPlacement.UNDEFINED` from that enumeration.
* Make `CENTER` the default for ELK Layered.

### PR DESCRIPTION
**Pull request by uruuru**
Fixes #666.

### CODE REVIEW

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java [hunk 1]
```diff
@@ -895,8 +892,7 @@ public static void configureWithDefaultValues(final ElkPort port) {
      *            an edge of a graph
      */
     public static void configureWithDefaultValues(final ElkEdge edge) {
-        EdgeLabelPlacement elp = edge.getProperty(CoreOptions.EDGE_LABELS_PLACEMENT);
-        if (elp == EdgeLabelPlacement.UNDEFINED) {
+        if (!edge.hasProperty(CoreOptions.EDGE_LABELS_PLACEMENT)) {
             edge.setProperty(CoreOptions.EDGE_LABELS_PLACEMENT, EdgeLabelPlacement.CENTER);
```
Comment (le-cds) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java:896]: Is this necessary given that the default value is `CENTER`?

File: plugins/org.eclipse.elk.conn.gmf/src/org/eclipse/elk/conn/gmf/GmfDiagramLayoutConnector.java [hunk 2]
```diff
@@ -885,7 +883,7 @@ protected void processConnections(final LayoutMapping mapping) {
         
         for (ConnectionEditPart connection : mapping.getProperty(CONNECTIONS)) {
             boolean isOppositeEdge = false;
-            EdgeLabelPlacement edgeLabelPlacement = EdgeLabelPlacement.UNDEFINED;
+            Optional<EdgeLabelPlacement> edgeLabelPlacement = Optional.empty();
```
Comment (le-cds) [inline plugins/org.eclipse.elk.conn.gmf/src/org/eclipse/elk/conn/gmf/GmfDiagramLayoutConnector.java:886]: Why not use `null` to indicate absence?

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java (hunk 1 above)
Comment (uruuru) [reply plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java:896]: As discussed, I didn't want to change the previous behavior: setting the value explicitly. We could change it in the future.

File: plugins/org.eclipse.elk.conn.gmf/src/org/eclipse/elk/conn/gmf/GmfDiagramLayoutConnector.java (hunk 2 above)
Comment (uruuru) [reply plugins/org.eclipse.elk.conn.gmf/src/org/eclipse/elk/conn/gmf/GmfDiagramLayoutConnector.java:886]: I prefer to use optionals in such cases to explicitly state that an absent value is allowed and expected.

**Pull request merged**